contracts/InterchainTokenFactory.sol

```diff
@@ -147,11 +139,9 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
             IInterchainToken token = IInterchainToken(service.interchainTokenAddress(tokenId));
             ITokenManager tokenManager = ITokenManager(service.tokenManagerAddress(tokenId));
 
-            _setDeployerTokenBalance(tokenId, sender, initialSupply);
-            token.mint(address(this), initialSupply);
+            token.mint(minter != address(0) ? minter : sender, initialSupply);
```

Comment:
```suggestion
            token.mint(sender, initialSupply);
```
`minter` might be a multisig or so, so maybe we send it to `sender` regardless, so they can move it to other chains right away. The sender wallet will already be connected to the portal.

Reply:
Fixed